### ISSUE TITLE
contrib/devtools/symbol-check.py gives many false-positive errors

### ISSUE THREAD

**Issue by daira** [zcash/zcash#4292]
### Describe the issue
`contrib/devtools/symbol-check.py` is supposed to check a binary (given as an argument) for unexpected symbols and dynamically linked libraries.

#### If so, please list the steps to reproduce below:
1. Build zcashd from source.
2. Run `contrib/devtools/symbol-check.py src/zcashd`

### Expected behaviour
No output.

### Actual behaviour + errors
Many false positives of the form:
```
src/zcashd: symbol std::... from unsupported version GLIBCXX_{3.4.15, 3.4.21, 3.4.22}
src/zcashd: symbol __... from unsupported version CXXABI_{1.3.5, 1.3.8, 1.3.9, 1.3.11}
src/zcashd: symbol __... from unsupported version GLIBC_{2.17, 2.18, 2.25}
src/zcashd: export of symbol ... not allowed
```

### The version of Zcash you were using:
```
Zcash Daemon version v2.1.1-rc1-2f8cf61a2
```

### Machine specs:
Debian Buster
```
$ uname -a
Linux katava 4.19.0-6-amd64 #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64 GNU/Linux
$ gcc --version
gcc (Debian 8.3.0-6) 8.3.0
[...]
$ ld -v
GNU ld (GNU Binutils for Debian) 2.31.1
$ as --version
GNU assembler (GNU Binutils for Debian) 2.31.1
```

**Comment by daira** [comment]
This gets rid of all the `symbol ... from unsupported version` errors:
```
diff --git a/contrib/devtools/symbol-check.py b/contrib/devtools/symbol-check.py
index 5b83ba97b..3f3446b32 100755
--- a/contrib/devtools/symbol-check.py
+++ b/contrib/devtools/symbol-check.py
@@ -39,9 +39,9 @@ import os
 #
 MAX_VERSIONS = {
 'GCC':     (4,4,0),
-'CXXABI':  (1,3,3),
-'GLIBCXX': (3,4,13),
-'GLIBC':   (2,11)
+'CXXABI':  (1,3,11),
+'GLIBCXX': (3,4,22),
+'GLIBC':   (2,25)
 }
 # See here for a description of _IO_stdin_used:
 # https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=634261#109
@@ -62,7 +62,6 @@ ALLOWED_LIBRARIES = {
 'libanl.so.1', # DNS resolve
 'libm.so.6', # math library
 'librt.so.1', # real-time (clock)
-'libgomp.so.1', # OpenMP support library
 'ld-linux-x86-64.so.2', # 64-bit dynamic linker
 'ld-linux.so.2', # 32-bit dynamic linker
 'libdl.so.2' # programming interface to dynamic linker
```
but not the `export of symbol ... not allowed` errors.

(Removing `libgomp.so.1` is not necessary, but we no longer depend on this library.)

**Comment by daira** [comment]
Note that there are some comments above the `MAX_VERSIONS` definition attempting to justify these version choices, which should also be updated. The intent is just that we don't use symbols that are only available from libraries that aren't packaged on older distributions that we still want to support.

**Comment by daira** [comment]
This is fixed by #4613 and https://github.com/str4d/zcash/pull/22 .
